renku/command/command_builder/command.py

```diff
@@ -171,34 +167,25 @@ def _injection_pre_hook(self, builder: "Command", context: dict, *args, **kwargs
             builder("Command"): Current ``CommandBuilder``.
             context(dict): Current context dictionary.
         """
-        from renku.command.command_builder.client_dispatcher import ClientDispatcher
-        from renku.core.interface.client_dispatcher import IClientDispatcher
-        from renku.core.management.client import LocalClient
-
-        dispatcher = ClientDispatcher()
-
         ctx = click.get_current_context(silent=True)
         if ctx is None:
-            if self._client:
-                dispatcher.push_created_client_to_stack(self._client)
-            else:
-                path = get_git_path(self._working_directory or ".")
-                project_context.push_path(path)
-                self._client = dispatcher.push_client_to_stack(path=path)
-                self._client_was_created = True
+            path = get_git_path(self._working_directory or ".")
+            project_context.push_path(path)
             ctx = click.Context(click.Command(builder._operation))  # type: ignore
+            self._context_was_created = True
         else:
-            if not self._client:
-                self._client = ctx.ensure_object(LocalClient)
-            dispatcher.push_created_client_to_stack(self._client)
+            # TODO: No path was pushed in the previous code. Is it true? Can this method be called more than once?
```

Comment:
but the `ClickContext.invoke` is only useful when calling a click command (like `renku run` calling `renku status` or something). But all our commands are plain python functions, so they don't need the context.

I think
```
output = self._operation(*args, **kwargs)
```

would work just as well instead of the current
```
with context["stack"]:
        output = context["click_context"].invoke(self._operation, *args, **kwargs)
```
Especially since anything not in `renku.ui.cli` shouldn't be using click anyways. You could give it a quick try and if it's more complicated, create a follow-up issue. I think this is just a left-over from when we had way worse separation between layers.